src/app/ipv/middleware.js

```diff
@@ -274,6 +275,11 @@ module.exports = {
             renderOptions.pageErrorState = req.query.errorState;
           }
 
+          if (req.session.currentPageStatusCode !== undefined) {
+            res.status(req.session.currentPageStatusCode);
+            delete req.session.currentPageStatusCode;
```

Comment:
Do you think we need to clear this anywhere else as well - e.g. if there's an unrecoverable timeout, or you end up on the wrong page for some reason?

It should never happen, but I wonder if we want to clear this status code on _any_ page request (e.g. using a `finally`)?

Reply:
Yeah, that's a good shout. I'll move it to a finally block.